test/OpenTelemetry.Exporter.Prometheus.HttpListener.Tests/PrometheusMetricTests.cs

```diff
@@ -0,0 +1,220 @@
+
+using Xunit;
+
+namespace OpenTelemetry.Exporter.Prometheus.Tests;
+
+public sealed class PrometheusMetricTests
+{
+    [Fact]
+    public void SanitizeMetricName_Valid()
+    {
+        AssertSanitizeMetricName("active_directory_ds_replication_network_io", "active_directory_ds_replication_network_io");
+    }
+
+    [Fact]
+    public void SanitizeMetricName_RemoveConsecutiveUnderscores()
+    {
+        AssertSanitizeMetricName("cpu_sp__d_hertz", "cpu_sp_d_hertz");
+    }
+
+    [Fact]
+    public void SanitizeMetricName_SupportLeadingAndTrailingUnderscores()
+    {
+        AssertSanitizeMetricName("_cpu_speed_hertz_", "_cpu_speed_hertz_");
+    }
+
+    [Fact]
+    public void SanitizeMetricName_RemoveUnsupportedChracters()
+    {
+        AssertSanitizeMetricName("metric_unit_$1000", "metric_unit_1000");
+    }
+
+    [Fact]
+    public void SanitizeMetricName_RemoveWhitespace()
+    {
+        AssertSanitizeMetricName("unit include", "unit_include");
+    }
+
+    [Fact]
+    public void SanitizeMetricName_RemoveMultipleUnsupportedChracters()
+    {
+        AssertSanitizeMetricName("sample_me%%$$$_count_ !!@unit include", "sample_me_count_unit_include");
+    }
+
+    [Fact]
+    public void SanitizeMetricName_RemoveStartingNumber()
+    {
+        AssertSanitizeMetricName("1_some_metric_name", "_some_metric_name");
+    }
+
+    [Fact]
+    public void SanitizeMetricName_SupportColon()
+    {
+        AssertSanitizeMetricName("sample_metric_name__:_per_meter", "sample_metric_name_:_per_meter");
+    }
+
+    [Fact]
+    public void Unit_Annotation_None()
+    {
+        Assert.Equal("Test", PrometheusMetric.RemoveAnnotations("Test"));
+    }
+
+    [Fact]
+    public void Unit_Annotation_RemoveLeading()
+    {
+        Assert.Equal("%", PrometheusMetric.RemoveAnnotations("%{percentage}"));
+    }
+
+    [Fact]
+    public void Unit_Annotation_RemoveTrailing()
+    {
+        Assert.Equal("%", PrometheusMetric.RemoveAnnotations("{percentage}%"));
+    }
+
+    [Fact]
+    public void Unit_Annotation_RemoveLeadingAndTrailing()
+    {
+        Assert.Equal("%", PrometheusMetric.RemoveAnnotations("{percentage}%{percentage}"));
+    }
+
+    [Fact]
+    public void Unit_Annotation_RemoveMiddle()
+    {
+        Assert.Equal("startend", PrometheusMetric.RemoveAnnotations("start{percentage}end"));
+    }
+
+    [Fact]
+    public void Unit_Annotation_RemoveEverything()
+    {
+        Assert.Equal(string.Empty, PrometheusMetric.RemoveAnnotations("{percentage}"));
+    }
+
+    [Fact]
+    public void Unit_Annotation_Multiple_RemoveEverything()
+    {
+        Assert.Equal(string.Empty, PrometheusMetric.RemoveAnnotations("{one}{two}"));
+    }
+
+    [Fact]
+    public void Unit_Annotation_NoClose()
+    {
+        Assert.Equal("{one", PrometheusMetric.RemoveAnnotations("{one"));
+    }
+
+    [Fact]
+    public void Unit_AnnotationMismatch_NoClose()
+    {
+        Assert.Equal("}", PrometheusMetric.RemoveAnnotations("{{one}}"));
```

Comment:
Shouldn't this result in an empty string?

Reply:
I don't know what it should do. The UCUM spec says curly braces shouldn't be nested but doesn't say what should happen if that happens.

https://ucum.org/ucum#section-Character-Set-and-Lexical-Rules

The current logic doesn't attempt to balance the number of start and end braces. That results in this output.